src/_http/http2Adapter.js

```diff
@@ -353,10 +353,11 @@ Http2Adapter.prototype.close = function(opts) {
  * @param {Error} error Error object.
  * @returns {Error} Remapped error.
  */
-function remapHttp2Error(error) {
+function remapHttp2Error({ error, isClosed }) {
   var shouldRemap =
-    error.code === 'ERR_HTTP2_GOAWAY_SESSION' ||
-    error.code === 'ERR_HTTP2_STREAM_CANCEL'
+    !isClosed &&
```

Comment:
We should remap to `ClientClosed` error only if `isClosed` is `true`:
```suggestion
    isClosed &&
```